geometry/optimization/iris.cc

```diff
@@ -522,6 +530,495 @@ std::vector<int> revolute_joint_indices(const multibody::Joint<double>& joint) {
+                      -binding.evaluator()->lower_bound(), binding.variables());
+        auto pos = std::find(additional_constraint_bindings.begin(),
+                             additional_constraint_bindings.end(), binding);
+        DRAKE_ASSERT(pos != additional_constraint_bindings.end());
+        additional_constraint_bindings.erase(pos);
+      }
+    };
+    HandleLinearConstraints(
+        options.prog_with_additional_constraints->bounding_box_constraints());
+    HandleLinearConstraints(
+        options.prog_with_additional_constraints->linear_constraints());
+    counter_example_prog = std::make_unique<CounterExampleProgram>(
+        counter_example_constraint, E, A.topRows(num_initial_constraints),
+        b.head(num_initial_constraints));
+
+    P_initial = HPolyhedron(A.topRows(num_initial_constraints),
+                    b.head(num_initial_constraints));
+  }
+
+  if (options.termination_func && options.termination_func(P_initial)) {
+    throw std::runtime_error(
+        "IrisInConfigurationSpace: The options.termination_func() returned "
+        "true for the initial region (defined by the linear constraints in "
+        "prog_with_additional_constraints and bounding_region arguments).  "
+        "Please check the implementation of your termination_func.");
+  }
+
+  DRAKE_THROW_UNLESS(P_initial.PointInSet(seed, 1e-12));
+  double best_volume = E.Volume();
+  int iteration = 0;
+  VectorXd closest(nq);
+  // RandomGenerator generator(options.random_seed);
+  RandomGenerator generator(random_seed);
+  std::vector<std::pair<double, int>> scaling(nc);
+  MatrixXd closest_points(nq, nc);
+
+  auto solver = solvers::MakeFirstAvailableSolver(
+      {solvers::SnoptSolver::id(), solvers::IpoptSolver::id()});
+
+
+  // auto mutable_context = plant.CreateDefaultContext();
+  // mutable_context->SetTimeStateAndParametersFrom(context);
+
+  VectorXd guess = seed;
+
+  const std::string seed_point_error_msg =
+      "IrisInConfigurationSpace: require_sample_point_is_contained is true but "
+      "the seed point exited the initial region. Does the provided "
+      "options.starting_ellipse not contain the seed point?";
+  const std::string seed_point_msg =
+      "IrisInConfigurationSpace: terminating iterations because the seed point "
+      "is no longer in the region.";
+  const std::string termination_error_msg =
+      "IrisInConfigurationSpace: the termination function returned false on "
+      "the computation of the initial region. Are the provided "
+      "options.starting_ellipse and options.termination_func compatible?";
+  const std::string termination_msg =
+      "IrisInConfigurationSpace: terminating iterations because "
+      "options.termination_func returned false.";
+
+  // auto query_object_mutable_context =
+  //         plant.get_geometry_query_input_port().template Eval<QueryObject<double>>(*mutable_context);
+  const Parallelism parallelism = Parallelism::Max();
```

Comment:
TODO: this should be an argument the user provides somewhere? Perhaps in IrisOptions?